### ISSUE TITLE
Reverse the order of `espanso log` output

### ISSUE THREAD

**Issue by woerndl** [federico-terzi/espanso#2081]
### Description

This is a minor change/issue, but still: I'd propose the output of the log should show the most recent events first. It makes little sense to scroll over the whole history, everytime you want to debug an error.

### Have you tried building it? Or would you like it?

- [ ] I'd be willing to contribute this feature to Espanso myself.

**Comment by smeech** [comment]
Good idea. I have amended your title a little. One would need to think about how much is shown, otherwise the "top" will soon scroll off-screen.

I find `tail -F ~/.cache/espanso/espanso.log | grep -v "INFO"` running in a terminal a useful way of following what's going on, without the routine messages, and occasionally use `tail -F ~/.cache/espanso/espanso.log`, both attached to Espanso triggers, of course!

**Comment by AucaCoyan** [comment]
Hey! thanks for submtting an issue to talk about this. 
Let me ask you, isn't it a subjective preference? I know we are tired of scrolling (that's why I don't use many log files, tends to be tedious over time).

We can change it, sure, but we have to know that we are changing a guideline for another guideline.
If you allow me, I prefer to have it stay the way it is. It just feels more natural with the "more commonly encountered" kind of file logs (at least what I read so far!)

**Comment by woerndl** [comment]
> Let me ask you, isn't it a subjective preference?

Of course, there's a subjective component to such behaviors. But from a UX perspective, Espanso (very!) often refers users to the log files for checking errors. Imagine you create a new snippet, perhaps something containing a script, and encounter regular errors throughout the process.

After every single error, you must open the log and scroll to the bottom just to see what is going on. That is tedious for no good reason and I'd argue that's objectively a poor experience.

Potential alternative solutions could be to a) introduce a setting b) automatically jump to the latest error when opening the log (or at least with a button in the interface) or c) to make error handling more intuitive overall. But for a project with limited resources, simply reversing the log order to make it more usable would be an easy fix.

**Comment by smeech** [comment]
I'm going to agree with @AucaCoyan on this one. The behaviour of `espanso log` is consistent with other programs. Like them, it's less than helpful, but there are command-line options like those I mentioned above to present log-output in other ways, and others which will reverse the output entirely for those who want.

I would suggest the real problem is the unhelpful nature of Espanso error messages.

**Comment by AucaCoyan** [comment]
You do have a point with unhelpful logs, I didn't have a problem as a dev because if something happens I jump to the code, (but that a dev error 🤣). Of these solutions:

> Potential alternative solutions could be to 
- a) introduce a setting 
- b) automatically jump to the latest error when opening the log (or at least with a button in the interface) or 
- c) to make error handling more intuitive overall. But for a project with limited resources, simply reversing the log order to make it more usable would be an easy fix.

A doesn't fit to me, I think we have too much configs and I would like to reduce them, some of them aren't working correctly and we have not the best overall defaults.
B this is not bad
C is the best in my opinion. Along with a purge of unhelful messages and add some needed ones (like some configs values smeechs asks over and over to users) we can also prettify the output of the logs, or show errors with [miette](https://github.com/zkat/miette?tab=readme-ov-file#about) and/or [human-panic](https://github.com/rust-cli/human-panic?tab=readme-ov-file#human-panic-output).

What do you think?

_gosh I love how GOOD is the user experience in rust, I wish I had these crates in my job with python_

**Comment by AucaCoyan** [comment]
BTW, I added to the project roadmap because better UX is going to make our product a lot better.
I want this!

**Comment by smeech** [comment]
The `tail`/`grep` command above supplies all my needs. 

I would just like better error-reporting during use, e.g. a popup containing the last error from the log.

I wouldn't mess with the log itself - it's familiar, and can be manipulated by external commands, but change the output of `espanso log` by all means.

**Comment by smeech** [comment]
#1261 related.
